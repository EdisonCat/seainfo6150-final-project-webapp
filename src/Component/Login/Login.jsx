import React, { useState, useContext } from "react";
import styles from "../SignUp/SignUp.module.css";
import MarvelFandomContext from "../../Context/MarvelFandomContext";
import { useHistory } from "react-router-dom";
export default function Login() {
	const [email, setEmail] = useState("");
	const [password, setPassword] = useState("");
	const context = useContext(MarvelFandomContext);
	const history = useHistory();
	const handleClick = (e) => {
		e.preventDefault();
		if (
			!localStorage.getItem("mf_email") ||
			localStorage.getItem("mf_email") !== email
		) {
			alert("We could NOT find your account");
			return;
		} else if (localStorage.getItem("mf_pass") !== password) {
			alert("Password is wrong");
			return;
		}
		context.login(localStorage.getItem("mf_username"));
		history.push({ pathname: "/" });
	};
	const handleRedirect = (e) => {
		e.preventDefault();
		history.push({ pathname: "/signup" });
	};
	return (
		<div className={styles.container}>
			<form>
				<input
					className={styles.inputtext}
					type="text"
					placeholder="E-mail"
					onChange={(e) => setEmail(e.target.value)}
				/>
				<input
					className={styles.inputtext}
					type="password"
					placeholder="Password"
					onChange={(e) => setPassword(e.target.value)}
				/>
			</form>
			<button onClick={(e) => handleClick(e)}>Login</button>
			<button onClick={(e) => handleRedirect(e)}>Register</button>
		</div>
	);
}
