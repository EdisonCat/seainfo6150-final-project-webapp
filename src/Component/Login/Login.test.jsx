import React from "react";
import Login from "./Login";
it("rendered Login correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
});