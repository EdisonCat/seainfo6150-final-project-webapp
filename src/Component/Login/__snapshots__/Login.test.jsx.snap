// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Login correctly 1`] = `
<div>
  <div
    class="container"
  >
    <form>
      <input
        class="inputtext"
        placeholder="E-mail"
        type="text"
      />
      <input
        class="inputtext"
        placeholder="Password"
        type="password"
      />
    </form>
    <button>
      Login
    </button>
    <button>
      Register
    </button>
  </div>
</div>
`;
